src/geometry.rs

```diff
@@ -27,7 +27,7 @@ pub fn compute_barycentric_coords(
     let ys = Vector3::new(ac.y, ab.y, pa.y);
     let ortho = xs.cross(&ys);
 
-    if f32::abs(ortho.z) < 1.0 {
+    if f32::abs(ortho.z) == 0.0 {
```

Comment:
One question remains... `==` vs `approx_eq`?

Reply:
you decide...

Reply:
I'd go for approx for floats instead of  `==` 99% of the time.